### PR TITLE
LCAM-304-New endpoint

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-304)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java [hunk 1]
```diff
@@ -81,4 +84,15 @@ public ResponseEntity<Object> getRepOrderMvoByRepIdAndVehicleOwner(@PathVariable
 
     }
 
+    @GetMapping(value = {"/{repId}/rep-order-count-with-sentence-order-date"}, produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description = "Retrieve a rep order record")
```
Comment (venkat980) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java:88]: Description is misleading, should this be Retrieve rep order count for a given sentence order date?

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java [hunk 2]
```diff
@@ -36,4 +36,11 @@ public void updateAppDateCompleted(final UpdateAppDateCompleted updateAppDateCom
         repOrderImpl.updateAppDateCompleted(updateAppDateCompleted.getRepId(), updateAppDateCompleted.getAssessmentDateCompleted());
         log.info("update app date completed  - Transaction Processing - End");
     }
+
+    @Transactional
+    public Boolean repOrderCountWithSentenceOrderDate(Integer repId) {
+        log.info("Get Rep order by appeal sentence order date is null - Transaction Processing - Start");
```
Comment (venkat980) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java:42]: Logging is misleading, should this be 'date is **NOT** null'?

**Review by venkat980 — COMMENTED** [review]
Looks good, except for couple of trivial comments. Thanks.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java (hunk 1 above)
Comment (loganathanMOJ) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java]: Thanks. Done now

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java (hunk 2 above)
Comment (loganathanMOJ) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java]: Thanks. Done now

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java [hunk 3]
```diff
@@ -85,8 +85,20 @@ public ResponseEntity<Object> getRepOrderMvoByRepIdAndVehicleOwner(@PathVariable
...
+    @ApiResponse(responseCode = "200", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE))
+    @ApiResponse(responseCode = "400", description = "Bad Request.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ErrorDTO.class)))
+    @ApiResponse(responseCode = "500", description = "Server Error.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ErrorDTO.class)))
+    public ResponseEntity<Object> updateRepOrder(@Parameter(description = "Update a rep order record", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = UpdateRepOrder.class))) @RequestBody UpdateRepOrder updateRepOrder) {
+        log.debug("Update Rep order request received for repId : {}", updateRepOrder.getRepId());
+        maatIdValidator.validate(updateRepOrder.getRepId());
+        repOrderService.updateRepOrder(updateRepOrder);
+        return ResponseEntity.ok().build();
+    }
+
```
Comment (Mhowell494) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/controller/RepOrderController.java:99]: Does this need to be a separate endpoint, could we not re-purpose the find endpoint and simply add a query string parameter to filter on the sentence order date? That seems much simpler.

I think the best solution here would be to use a `HEAD` request on the same URL for retrieving repOrders that returns the count metadata as a header.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

File: maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java [hunk 4]
```diff
@@ -38,10 +38,19 @@ public void updateAppDateCompleted(final UpdateAppDateCompleted updateAppDateCom
...
+        log.info("update rep order - Transaction Processing - Start");
+        RepOrderEntity repOrderEntity = repOrderImpl.find(updateRepOrder.getRepId());
+        repOrderMapper.updateRepOrderToRepOrderEntity(updateRepOrder, repOrderEntity);
+        repOrderImpl.updateRepOrder(repOrderEntity);
+        log.info("update rep order  - Transaction Processing - End");
+    }
+
     @Transactional
     public Boolean repOrderCountWithSentenceOrderDate(Integer repId) {
-        log.info("Get Rep order by appeal sentence order date is null - Transaction Processing - Start");
```
Comment (Mhowell494) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/service/RepOrderService.java]: I don't think it makes sense to include the word count in the method name if you're returning a boolean. Something like `hasRepOrderWithSentenceOrderDate` would be a lot more descriptive

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

**Review by venkat980 — APPROVED** [review]

**Review by Mhowell494 — APPROVED** [review]

**Pull request merged**